include/range/v3/utility/any.hpp

```diff
@@ -124,7 +124,7 @@ namespace ranges
         any() noexcept = default;
         template(typename TRef, typename T = detail::decay_t<TRef>)(
             /// \pre
-            requires copyable<T> AND (!same_as<T, any>)) //
+            requires copy_constructible<T> AND (!same_as<T, any>)) //
```

Comment:
I think if the `!same_as<...>` check came before the `copyable` check, that would fix the problem.